FileMapper/API_util.js

```diff
@@ -0,0 +1,307 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * API Utilities
+ */
+var ApiUtil = {
+  checkInConfig: checkInConfig,
+  getFromConfig: getFromConfig,
+  addNewMappingToConfig: addNewMappingToConfig,
+  checkIfMarkedDeleted: checkIfMarkedDeleted,
+  findInDrive: findInDrive,
+  getDestinationId: getDestinationId,
+  createDeletedDestination: createDeletedDestination,
+  createInDrive: createInDrive,
+  getRegExFromPattern: getRegExFromPattern
+}
+
+/**
+ * Checks if the mapping for a particular Local path exists or not in the config
+ * 
+ * @param {String} localPath The local destination path whose mapping is to be checked
+ * @return {boolean} True if mapping exists, 
+ *                   False otherwise
+ */
+function checkInConfig(localPath) {
+  var documentProperties = PropertiesService.getDocumentProperties();
+  var value = documentProperties.getProperty(localPath);
+  return (value !== null);
+}
+
+/**
+ * Get the mapping corresponding to the localPath
+ *
+ * @param {String} localPath The local destination path whose mapping is to be returned
+ * @return {String} value The corresponding drive Id if mapping exists, 
+ *                  null otherwise
+ */
+function getFromConfig(localPath) {
+  var documentProperties = PropertiesService.getDocumentProperties();
+  var value = documentProperties.getProperty(localPath);
+  if (value !== null) {
+    var mappingObj = JSON.parse(value);
+    return mappingObj.id;
+  }
+  else {
+    return null;
+  }
+}
+
+/**
+ * Adds a mapping to the config 
+ * 
+ * @param {String} localPath The local destination path
+ * @param {String} id The corresponding drive destination Id
+ * @param {boolean} isFile To signify whether its a file or folder
+ */
+function addNewMappingToConfig(localPath, id, isFile) {
+  var documentProperties = PropertiesService.getDocumentProperties();
+  var drivepath = getFullDrivePath(id, isFile);
+  var mappingObject = {
+    id: id,
+    drivepath: drivepath,
+    isfolder: !isFile
+  };
+  documentProperties.setProperty(localPath, JSON.stringify(mappingObject));
```

Comment:
Are we storing individual  key-value pairs? How will we distinguish this with whatever key-value pairs user might add to the document properties?

Reply:
These are the key-value pairs being found when we perform getDriveId which has been generated based on mappings only. Yes, I am storing key-pairs directly to document properties.
What exactly are you asking?

Reply:
Do you mean that i should store all the config data as one key-value pair?

Reply:
Yes. Otherwise let's say we keep in the same way and user adds his own key-value pairs like locale: "EN-US"

Then when reading our config how can we make sure that this is not something we added? So if we instead store the whole JSON as a value with a key like localFileDriveMapping then there is no issue with whatever other keys being stored by the user?

Reply:
Yeah, that was one of the designs i also thought about initially.
I will update the design and will save the config data as one key-value pair.